pkg/devfile/adapters/docker/component/utils.go

```diff
@@ -344,77 +344,61 @@ func getPortMap(context string, endpoints []*versionsCommon.Endpoint, show bool)
 
 // Executes all the commands from the devfile in order: init and build - which are both optional, and a compulsary run.
 // Init only runs once when the component is created.
-func (a Adapter) execDevfile(pushDevfileCommands []versionsCommon.DevfileCommand, componentExists, show bool, containers []types.Container) (err error) {
+func (a Adapter) execDevfile(commandsMap common.CommandsMap, componentExists, show bool, containers []types.Container) (err error) {
 	// If nothing has been passed, then the devfile is missing the required run command
-	if len(pushDevfileCommands) == 0 {
+	if len(commandsMap) == 0 {
 		return errors.New(fmt.Sprint("error executing devfile commands - there should be at least 1 command"))
 	}
 
-	commandOrder := []common.CommandNames{}
-
 	// Only add runinit to the expected commands if the component doesn't already exist
 	// This would be the case when first running the container
 	if !componentExists {
-		commandOrder = append(commandOrder, common.CommandNames{DefaultName: string(common.DefaultDevfileInitCommand), AdapterName: a.devfileInitCmd})
+		// Get Init Command
+		command, ok := commandsMap[versionsCommon.InitCommandGroupType]
+		if ok {
+
+			containerID := utils.GetContainerIDForAlias(containers, command.Exec.Component)
```

Comment:
Should we rename this function? Because Alias is not there in v2 and this seems confusing to me.

Reply:
yes, we should rename, not doing it as a part of this PR.

Reply:
OK